docs/user/reference/resource-compute/list-of-commands-available-on-a-compute-resource-provisioned-by-juju/list-of-juju-introspect-macros/juju_controller_agent_name.md

```diff
@@ -1,4 +1,22 @@
 (juju_controller_agent_name)=
 # `juju_controller_agent_name`
 
-TBA
+The `juju_controller_agent_name` macro returns the name of the controller 
+agent that is currently running.
+
+## Usage
+
+```python
+juju_controller_agent_name
+```
+
+Returns a string containing the agent (controller) name.
+
+_Note: This macro is only available on controller machines, everywhere else
```

Comment:
```suggestion
\```{note}
This macro is only available on controller machines -- everywhere else it returns an empty string.
\```
```